prospero-common/src/main/java/org/wildfly/prospero/galleon/MavenArtifactDirectResolverRepositoryManager.java

```diff
@@ -0,0 +1,122 @@
+ * limitations under the License.
+ */
+
+package org.wildfly.prospero.galleon;
+
+import org.jboss.galleon.universe.maven.MavenArtifact;
+import org.jboss.galleon.universe.maven.MavenUniverseException;
+import org.jboss.galleon.universe.maven.repo.MavenRepoManager;
+import org.wildfly.channel.ChannelSession;
+import org.wildfly.channel.UnresolvedMavenArtifactException;
+import org.wildfly.channel.spi.ChannelResolvable;
+
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * A MavenRepoManager implementation that will do direct artifact resolving.
+ * This works only on MavenArtifact resolving whose version is known.
+ */
+public class MavenArtifactDirectResolverRepositoryManager implements MavenRepoManager, ChannelResolvable {
+
+    private final ChannelSession channelSession;
+
+    public MavenArtifactDirectResolverRepositoryManager(ChannelSession channelSession) {
+        this.channelSession = channelSession;
+    }
+
+    @Override
+    public void resolve(MavenArtifact artifact) throws MavenUniverseException {
+        checkArtifact(artifact);
+        org.wildfly.channel.MavenArtifact result;
+        try {
+            result = channelSession.resolveDirectMavenArtifact(artifact.getGroupId(), artifact.getArtifactId(),
+              artifact.getExtension(), artifact.getClassifier(), artifact.getVersion());
+        } catch (UnresolvedMavenArtifactException ex) {
+            // if the artifact can not be resolved directly either, we abort
+            throw new MavenUniverseException(ex.getLocalizedMessage(), ex);
+        }
+        MavenArtifactMapper.resolve(artifact, result);
+    }
+
+    private void checkArtifact(MavenArtifact artifact) {
+        if (artifact.getVersion() == null) {
+            throw new IllegalArgumentException("Version of " + artifact + " is null");
+        }
+    }
+
+    @Override
+    public void resolveAll(Collection<MavenArtifact> artifacts) throws MavenUniverseException {
+        artifacts.forEach(this::checkArtifact);
+        MavenArtifactMapper mapper = new MavenArtifactMapper(artifacts);
+        List<org.wildfly.channel.MavenArtifact> channelArtifacts = channelSession.resolveDirectMavenArtifacts(mapper.toChannelArtifacts());
+        mapper.applyResolution(channelArtifacts);
+    }
+
+    @Override
+    public boolean isResolved(MavenArtifact artifact) throws MavenUniverseException {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    @Override
+    public boolean isLatestVersionResolved(MavenArtifact artifact, String lowestQualifier) throws MavenUniverseException {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    @Override
+    public void resolveLatestVersion(MavenArtifact artifact, String lowestQualifier, Pattern includeVersion, Pattern excludeVersion) throws MavenUniverseException {
+        resolve(artifact);
```

Comment:
I'm not sure about `resolveLatestVersion`. In case of a channel resolution this is the same as `resolve`, but when we're falling back to default resolution, we might need to get the latest version from Maven.

On the other hand, AFAICT that's only used to resolve MavenUniverse stuff which we don't expect to use